src/native/libs/System.Native/pal_datetime.c

```diff
@@ -67,7 +67,9 @@ const char* SystemNative_GetTimeZoneData(const char* name, int* length)
     assert(name != NULL);
     assert(length != NULL);
 #if defined(TARGET_WASI) || defined(TARGET_BROWSER)
-    return (const char*) mono_wasm_get_bundled_file(name, length);
+    unsigned char *data = NULL;
+    mono_bundled_resources_get_data_resource_values (name, &data, length);
```

Comment:
int * and uint32_t * are incompatible types. Instead do:

`#include <limits.h>`

...

```
const uint8_t *data = NULL;
uint32_t data_len = 0;

mono_bundled_resources_get_data_resource_values (name, &data, &data_len);
assert (data_len <= INT_MAX);
if (data_len > INT_MAX) {
    *data_len= 0;
    *data = NULL;
}

*length = (int)data_len;
return (const char *)data;
```

Reply:
Ah, I see, so if on a particular platform if the `int` is smaller than 32 bits then because the caller from managed side builds a copy of the data, it wouldn't be able to copy all of the information if the data length exceeds the size of an `int`. Thanks!